core/deployment/src/main/java/io/quarkus/deployment/BuildInfo.java

```diff
@@ -0,0 +1,48 @@
+package io.quarkus.deployment;
+
+public class BuildInfo {
+
+    private final String group;
+    private final String name;
+    private final String version;
+    private final String finalName;
+    private final String baseDir;
+    private final String wiringClassesDir;
+
+    public BuildInfo(String group, String name, String version, String finalName, String baseDir, String wiringClassesDir) {
+        this.group = group;
+        this.name = name;
+        this.version = version;
+        this.finalName = finalName;
+        this.baseDir = baseDir;
+        this.wiringClassesDir = wiringClassesDir;
+    }
+
+    public String getGroup() {
+        return group;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public String getVersion() {
+        return version;
+    }
+
+    public String getFinalName() {
+        return finalName;
+    }
+
+    public String getBaseDir() {
+        return baseDir;
+    }
+
+    public String getWiringClassesDir() {
+        return wiringClassesDir;
+    }
+
+    public static BuildInfo unset() {
```

Comment:
I think this should be a constant instead of a method. It's named as a verb so we expect some action.

BTW, why the empty strings instead of null? And don't you always expect it to be defined?

Reply:
That's a good point, I moved it to a static final. As for the avoidance of null, just a personal preference I guess. Would you like that I change it?